tests/IntegralTest.cpp

```diff
@@ -70,41 +60,38 @@ int main()
   config.debug = true;
 
   sql::connection db(config);
-  db.execute(R"(CREATE TABLE fp_sample (
-      id INTEGER,
-      fp REAL
+  db.execute(R"(CREATE TABLE integral_sample (
+      signed_value INTEGER,
+      unsigned_value INTEGER
   ))");
 
-  // check for automatic conversion in serializer
-  uint64_t v = 17032080461028570721ULL;
-  auto v2 = static_cast<int64_t>(v);
+  // Supported range for unsigned value (as int64 is the maximum)
```

Comment:
I might be wrong, but I thought the conversion is defined by the standard, in other words it should also work for `std::numeric_limits<uint64_t>::max()`?

If that is not guaranteed by the standard, can you point me to the relevant section?

Reply:
I'm not entirely sure, what you mean here. But I suppose my comment above line 69 might have confused you. What I meant is if you put anything bigger than int64_t in the lib it will still work (see line 75) technically, but the data in the database will be wrong (at least if you don't know that it's an uint64_t).
So `std::numeric_limits<uint64_t>::max()` will be converted (in a definied way as far as I know) into int64_t and therefore there will be an integer overflow resulting in -1.
The test shows that it still works (i.e. you get in the end std::numeric_limits<uint64_t>::max() back), but the database will save it as -1 and therefore another lib (or a human), who don't know that this is an unsigned value will interpret it wrong. That's want I meant by unsupported.

Reply:
I see what you mean. Maybe add another line or two as explanation?

Reply:
Is the documentation I just added clear enough? Do you miss anything?